gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -82,42 +94,138 @@ public synchronized ILiferayProject provide( Object adaptable )
     @Override
     public IStatus createNewProject( NewLiferayModuleProjectOp op, IProgressMonitor monitor ) throws CoreException
     {
-        IStatus retval = null;
+        IStatus retval = Status.OK_STATUS;
 
         final String projectName = op.getProjectName().content();
 
         IPath location = PathBridge.create( op.getLocation().content() );
 
+        String className = op.getComponentName().content();
+
+        final String serviceName = op.getServiceName().content();
+
+        final String packageName = op.getPackageName().content();
+
+        ElementList<PropertyKey> propertyKeys = op.getPropertyKeys();
+
+        final List<String> properties = new ArrayList<String>();
+
+        for( PropertyKey propertyKey : propertyKeys )
+        {
+            properties.add( propertyKey.getKeyName() + "=" + propertyKey.getKeyValue() );
+        }
+
         final String projectTemplateName = op.getProjectTemplateName().content();
 
         StringBuilder sb = new StringBuilder();
         sb.append( "create " );
-        sb.append( "-d \"" + location.toFile().getAbsolutePath() +  "\" " );
+        sb.append( "-d \"" + location.toFile().getAbsolutePath() + "\" " );
         sb.append( "-t " + projectTemplateName + " " );
-        sb.append( "\"" + projectName + "\"");
 
-        try
+        if( className != null )
         {
-            BladeCLI.execute( sb.toString() );
+            sb.append( "-c " + className + " " );
         }
-        catch( BladeCLIException e )
+
+        if( serviceName != null )
         {
-            retval = ProjectCore.createErrorStatus( e );
+            sb.append( "-s " + serviceName + " " );
         }
 
-        if( retval.isOK() )
+        if( packageName != null )
         {
-            ProjectImportConfiguration configuration = new ProjectImportConfiguration();
-            GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
-            configuration.setGradleDistribution( from );
-            configuration.setProjectDir( location.toFile() );
-            configuration.setApplyWorkingSets( false );
-            configuration.setWorkingSets( new ArrayList<String>() );
-            new SynchronizeGradleProjectJob(
-                configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
-                AsyncHandler.NO_OP ).schedule();
+            sb.append( "-p " + packageName + " " );
         }
 
+        sb.append( "\"" + projectName + "\" " );
+
+        try
+        {
+            final String[] ret = BladeCLI.execute( sb.toString() );
```

Comment:
Add some checking for errors in the blade tool.  (search for lines that start with "Error") because that is what was happening to me and it wasn't showing in the UI.